presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadataFactory.java

```diff
@@ -143,7 +143,7 @@ public HiveMetadata get()
     {
         SemiTransactionalHiveMetastore metastore = new SemiTransactionalHiveMetastore(
                 hdfsEnvironment,
-                CachingHiveMetastore.memoizeMetastore(this.metastore, perTransactionCacheMaximumSize), // per-transaction cache
+                this.metastore,
```

Comment:
Why a change here?

Reply:
Because this.metastore used to be evaluated as GlueHiveMetastore and everytime a new cache map was being created because of memoizeMetastore so it never read from cache, it always used to boil down to: `io.prestosql.plugin.hive.metastore.glue.GlueHiveMetastore#getTable`